File: portal/src/main/java/net/ownportal/portal/story/Favourite.java
package net.ownportal.portal.story;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.Getter;
import lombok.Setter;
import net.ownportal.portal.user.User;

@Entity
@Table(name = "favourites")
@Getter
@Setter
public class Favourite {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    @ManyToOne
    private User user;
    private String title;
    private String description;
    private String publishedDate;
    private String publisher;
    private String link;
}

File: api/app/models/favourite.rb
# == Schema Information
#
# Table name: favourites
#
#  id             :integer          not null, primary key
#  description    :string
#  link           :string
#  published_date :datetime
#  publisher      :string
#  title          :string
#  user_id        :integer
#  created_at     :datetime         not null
#  updated_at     :datetime         not null
#  tags_json      :text
#
class Favourite < ApplicationRecord
  has_many :taggings, dependent: :destroy
  has_many :tags, through: :taggings
  belongs_to :user
end
